server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -257,7 +265,7 @@ public RemoteClusterStateService(
         this.remoteClusterStateCleanupManager = new RemoteClusterStateCleanupManager(this, clusterService);
         this.indexMetadataUploadListeners = indexMetadataUploadListeners;
         this.remoteRoutingTableService = isRemoteRoutingTableEnabled(settings)
-            ? Optional.of(new RemoteRoutingTableService(repositoriesService, settings))
+            ? Optional.of(new RemoteRoutingTableService(repositoriesService, settings, clusterSettings))
```

Comment:
Instead of Optional `RemoteRoutingTableService ` and checking for `Optional#ispresent` all over the place, it would simplify if used a factory with a default `NoOpRoutingTableService`

Reply:
Changed to using a factory and noop impl.